<launch>
  <arg name="usb_port"                default="/dev/ttyUSB_dynamixel"/>
  <arg name="dxl_baud_rate"           default="1000000"/>
  <arg name="namespace"               default="manipulator"/>

  <arg name="use_moveit"              default="true"/>
  <arg name="use_joint_state"         default="true"/>
  <arg name="use_cmd_vel"             default="false"/>

  <arg name="dynamixel_info"          default="$(find aerial_manipulators_control)/config/icl_dynamixel_info.yaml"/>

  <arg name="gripper_aperture_topic"          default="/manipulator/icl_gripper_controller/gripper_aperture"/>
  
  <param name="$(arg namespace)/dynamixel_info"        value="$(arg dynamixel_info)"/>

  <node name="dynamixel_controller" pkg="dynamixel_workbench_controllers" type="dynamixel_workbench_controllers" ns="$(arg namespace)"
        required="true" args="$(arg usb_port) $(arg dxl_baud_rate)">
    <param name="use_moveit"              value="$(arg use_moveit)"/>
    <param name="use_joint_states_topic"  value="$(arg use_joint_state)"/>
    <param name="use_cmd_vel_topic"       value="$(arg use_cmd_vel)"/>
    <rosparam>
      publish_period: 0.010
      dxl_read_period: 0.010
      dxl_write_period: 0.010
      mobile_robot_config:                <!--this values will be set when 'use_cmd_vel' is true-->
        seperation_between_wheels: 0.160  <!--default value is set by reference of TB3-->
        radius_of_wheel: 0.033            <!--default value is set by reference of TB3-->
    </rosparam>
  </node>

  <!--node name="manipulator_control_node" pkg="aerial_manipulators_control" type="manipulator_control_node" output="screen" ns="$(arg namespace)">
        <param name="rate" value="30"/>
        <param name="robot_name" value="icl_manipulator"/>
        <param name="joint_group_name" value="icl_manipulator_gripper"/>
        <param name="parameters_file" value="/config/icl_manipulator_parameters.yaml"/>
  </node-->

  <node name="simple_dynamixel_controller" pkg="aerial_manipulators_control" type="simple_dynamixel_controller.py" output="screen" ns="$(arg namespace)">
    <param name="config_file" value="$(arg dynamixel_info)"/>
    <remap from="simple_dynamixel_controller/joint_trajectory_out" to="dynamixel_controller/joint_trajectory"/>
  </node>

  <node name="icl_gripper_controller" pkg="aerial_manipulators_control" type="icl_gripper_controller.py" output="screen" ns="$(arg namespace)">
    <param name="max_angle" value="1.5"/>
    <remap from="icl_gripper_controller/joint1/command" to="joint1/command"/>
    <remap from="icl_gripper_controller/joint2/command" to="joint2/command"/>
    <remap from="icl_gripper_controller/gripper_aperture" to="$(arg gripper_aperture_topic)"/>
  </node>

</launch>
